docs: add workaround for TS custom commands (#5454)
cypress-io/cypress-documentation: docs/api/cypress-api/custom-commands.mdx

[`cypress-example-todomvc`](https://github.com/cypress-io/cypress-example-todomvc#cypress-intellisense)
repository for a working example.

#### 6. Create a function that adds the custom command

> From cypress@12.17.4, if you are using TypeScript and have `sideEffects:false` set in your package.json, custom Cypress commands will not be registered by the common pattern of writing the commands in one file and having them be run as a side effect of importing the file. For example:

Reviewer: The "@" actually creates a `mailto:` link in the markdown.

```suggestion
From version [12.17.4](/guides/references/changelog#12-17-4), if you are using TypeScript and have `sideEffects:false` set in your package.json, custom Cypress commands will not be registered by the common pattern of writing the commands in one file and having them be run as a side effect of importing the file. For example:
```

> From cypress@12.17.4, if you are using TypeScript and have `sideEffects:false` set in your package.json, custom Cypress commands will not be registered by the common pattern of writing the commands in one file and having them be run as a side effect of importing the file. For example:
> 
> ```typescript title="/cypress/support/commands.ts"
> Cypress.Commands.add("login", (email, password) => { ... })
> ```
> 
> ```typescript title="/cypress/support/e2e.ts"
> import './commands'
> ```
> 
> To work around this issue, you can wrap the Cypress commands in a function that will be imported by the support file.
> 
> ```typescript title="/cypress/support/commands.ts"
> export function registerCommands(){
>   Cypress.Commands.add("login", (email, password) => { ... })
> }
> ```
> 
> ```typescript title="/cypress/support/e2e.ts"
> import { registerCommands } from './commands'
> registerCommands()
> ```

Reviewer: The "@" actually creates a `mailto:` link in the markdown. Here are some edits to give more context

```suggestion
Cypress [12.17.4](/guides/references/changelog#12-17-4) includes a webpack upgrade (v4 to v5), which [tree shakes out](https://webpack.js.org/blog/2020-10-10-webpack-5-release/#inner-module-tree-shaking) any side-effects or files that only include side-effects.

If you are using TypeScript and have `sideEffects:false` set in your package.json, custom Cypress commands will not be registered by the common pattern of writing the commands in one file and having them be run as a side effect of importing the file. For example:

```typescript
// cypress/support/commands.ts

Cypress.Commands.add("login", (email, password) => { ... })
```

```typescript
// cypress/support/e2e.ts

import './commands'
```

To support `sideEffects:false`, you can wrap the Cypress commands in a function that will be imported by the support file.

```typescript
// cypress/support/commands.ts

export function registerCommands(){
  Cypress.Commands.add("login", (email, password) => { ... })
}
```

```typescript
// cypress/support/e2e.ts

import { registerCommands } from './commands'
registerCommands()
```
```

[`cypress-example-todomvc`](https://github.com/cypress-io/cypress-example-todomvc#cypress-intellisense)
repository for a working example.

#### 6. Create a function that adds the custom command

> Cypress [12.17.4](/guides/references/changelog#12-17-4) includes a webpack upgrade (v4 to v5), which [tree shakes out](https://webpack.js.org/blog/2020-10-10-webpack-5-release/#inner-module-tree-shaking) any side-effects or files that only include side-effects.
> 
> If you are using TypeScript and have `sideEffects:false` set in your package.json, custom Cypress commands will not be registered by the common pattern of writing the commands in one file and having them be run as a side effect of importing the file. For example:
> 
> ```typescript title="/cypress/support/commands.ts"
> Cypress.Commands.add("login", (email, password) => { ... })
> ```

Reviewer: Weird that the markdown is breaking in the suggestion. Here are the rest of the original batch, plus a couple tiny improvements. 

```suggestion
Cypress [12.17.4](/guides/references/changelog#12-17-4) includes a Webpack upgrade (v4 to v5), which [tree shakes out](https://webpack.js.org/blog/2020-10-10-webpack-5-release/#inner-module-tree-shaking) any side-effects or files that only include side-effects.

If you are using TypeScript and have Webpack's `sideEffects:false` set in your package.json, custom Cypress commands will not be registered by the common pattern of writing the commands in one file and having them be run as a side effect of importing the file. For example:

```typescript
// cypress/support/commands.ts

Cypress.Commands.add("login", (email, password) => { ... })
```

> ```typescript title="/cypress/support/e2e.ts"
> import './commands'
> ```

Reviewer: Weird that the markdown is breaking in the suggestion. Here are the rest of the original batch, plus a couple tiny improvements. 

```suggestion

```
// cypress/support/e2e.ts

import './commands'
```

> To support `sideEffects:false`, you can wrap the Cypress commands in a function that will be imported by the support file.
> 
> ```typescript title="/cypress/support/commands.ts"
> export function registerCommands(){
>   Cypress.Commands.add("login", (email, password) => { ... })
> }
> ```

Reviewer: ```suggestion
To support `sideEffects:false`, you can wrap the Cypress commands in a function that will be imported by the support file.

```typescript
// cypress/support/commands.ts

export function registerCommands(){
  Cypress.Commands.add("login", (email, password) => { ... })
}
```

```

> ```typescript title="/cypress/support/e2e.ts"
> import { registerCommands } from './commands'
> registerCommands()
> ```

Reviewer: ```suggestion
```typescript
// cypress/support/e2e.ts

import { registerCommands } from './commands'
registerCommands()
```
